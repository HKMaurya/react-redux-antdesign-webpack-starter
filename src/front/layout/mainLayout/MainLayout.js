// @flow
/* eslint-disable react/no-unescaped-entities */

// #region imports
import React, { Component } from 'react';
import { Layout, Menu, Icon } from 'antd';
import navigationModel from '../../config/navigation.json';
import MainRoutes from '../../routes/MainRoutes';
import { type RouterProps } from '../../types/react-router';
import StyledLayout from './styled/StyledLayout';
import SideMenuLogo from './styled/SideMenuLogo';
// #endregion

// #region flow types
export type Props = { ...any } & RouterProps;

export type State = {
  navModel: navigationModel,
  selectedSidemenu: ['/'],
  ...any,
};
// #endregion

// #region constants
const { Header, Content, Footer, Sider } = Layout;
const MenuItem = Menu.Item; // workaround to fix production bundle error: "Menu not found"
// #endregion

class MainLayout extends Component<Props, State> {
  state = {
    navModel: navigationModel,
    selectedSidemenu: ['/'],
  };

  // #region lifecycle
  render() {
    const { navModel, selectedSidemenu } = this.state;

    return (
      <StyledLayout>
        <Sider
          breakpoint="lg"
          collapsedWidth="0"
          onCollapse={this.handlesOnCollpase}
        >
          <SideMenuLogo />
          <Menu
            theme="dark"
            mode="inline"
            defaultSelectedKeys={selectedSidemenu}
            onClick={this.handlesOnMenuClick}
          >
            {navModel.sideLinks.map(({ label, icon, link }) => (
              <MenuItem key={link}>
                <Icon type={icon} />
                <span className="nav-text">{label}</span>
              </MenuItem>
            ))}
          </Menu>
        </Sider>
        <Layout>
          <Header style={{ background: '#4A4A4A', padding: 0 }} />
          <Content style={{ margin: '24px 16px 0' }}>
            <div style={{ padding: 24, background: '#fff', minHeight: 360 }}>
              <MainRoutes />
            </div>
          </Content>
          <Footer style={{ textAlign: 'center' }}>
            "React+ Redux+ Ant Design + Webpack Starter" made with{' '}
            <span style={{ color: '#CF000F' }}>❤️</span> by Erwan Datin
          </Footer>
        </Layout>
      </StyledLayout>
    );
  }
  // #endregion

  // #region side menu collapse event
  handlesOnCollpase = (collapsed: boolean, type: string): void => {
    /* eslint-disable no-console */
    console.log(collapsed, type);
    /* eslint-enable no-console */
  };
  // #endregion

  // #region side menu item click event
  handlesOnMenuClick = (event: SyntheticEvent<>): void => {
    if (event) {
      const { history } = this.props;
      // $FlowIgnore
      const { key } = event;

      history.push(key);
    }
  };
  // #endregion
}

export default MainLayout;
