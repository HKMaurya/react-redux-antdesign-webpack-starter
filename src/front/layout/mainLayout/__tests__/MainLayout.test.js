// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import MainLayout from '../MainLayout';
import navigationModel from '../../../config/navigation.json';
// #endregion

// #region jest mocks
jest.mock('../../../routes/routes', () => ({
  Home: () => <p>home</p>,
  About: () => <p>home</p>,
  Protected: () => <p>home</p>,
  PrivateRoute: () => <p>home</p>,
}));
// #endregion

describe('MainLayout component', () => {
  const props = {
    navModel: navigationModel,
    selectedSidemenu: ['/'],
  };

  it('renders as expected', () => {
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <MainLayout {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
