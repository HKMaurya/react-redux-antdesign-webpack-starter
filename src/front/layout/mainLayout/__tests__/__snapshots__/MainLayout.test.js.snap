// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout component renders as expected 1`] = `
<div>
  <div
    className="sc-bdVaJa fORvkZ ant-layout"
  >
    <div
      className="ant-layout-header"
      style={
        Object {
          "padding": 0,
        }
      }
    />
    <div
      className="ant-layout ant-layout-has-sider"
    >
      <div
        className="ant-layout-sider ant-layout-sider-light"
        style={
          Object {
            "flex": "0 0 200px",
            "maxWidth": "200px",
            "minWidth": "200px",
            "width": "200px",
          }
        }
      >
        <div
          className="ant-layout-sider-children"
        >
          <div
            className="sc-bwzfXH hfBJpx"
          />
          <ul
            className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
            role="menu"
            style={Object {}}
          >
            <li
              className="ant-menu-item ant-menu-item-selected"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <i
                className="anticon anticon-home"
              />
              <span
                className="nav-text"
              >
                Home
              </span>
            </li>
            <li
              className="ant-menu-item"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <i
                className="anticon anticon-lock"
              />
              <span
                className="nav-text"
              >
                Protected
              </span>
            </li>
            <li
              className="ant-menu-item"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <i
                className="anticon anticon-info-circle-o"
              />
              <span
                className="nav-text"
              >
                About
              </span>
            </li>
            <li
              className="ant-menu-item"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="menuitem"
              style={
                Object {
                  "paddingLeft": 24,
                }
              }
            >
              <i
                className="anticon anticon-logout"
              />
              <span
                className="nav-text"
              >
                Disconnect
              </span>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="ant-layout"
      >
        <div
          className="ant-layout-content"
          style={
            Object {
              "margin": "24px 16px 0",
            }
          }
        >
          <div
            style={
              Object {
                "background": "#fff",
                "minHeight": 360,
                "padding": 24,
              }
            }
          >
            <p>
              home
            </p>
          </div>
        </div>
        <div
          className="ant-layout-footer"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          "React+ Redux+ Ant Design + Webpack Starter" made with
           
          <span
            style={
              Object {
                "color": "#CF000F",
              }
            }
          >
            ❤️
          </span>
           by Erwan Datin
        </div>
      </div>
    </div>
  </div>
</div>
`;
